systemtest/src/test/java/io/strimzi/systemtest/backup/ColdBackupScriptST.java

```diff
@@ -0,0 +1,123 @@
+        String backupFilePath = USER_PATH + "/target/" + clusterName + ".zip";
+
+        // deploy a test cluster
+        installClusterOperator(NAMESPACE);
+        LOGGER.info("Deploying test cluster {}", clusterName);
+        KafkaResource.createAndWaitForReadiness(KafkaResource.kafkaPersistent(clusterName, 1, 1).build());
+        String clientsPodName = deployAndGetInternalClientsPodName();
+        InternalKafkaClient clients = new InternalKafkaClient.Builder()
+            .withListenerName(Constants.PLAIN_LISTENER_DEFAULT_NAME)
+            .withNamespaceName(NAMESPACE)
+            .withUsingPodName(clientsPodName)
+            .withClusterName(clusterName)
+            .withTopicName(topicName)
+            .withConsumerGroupName(groupName)
+            .withMessageCount(firstBatchSize)
+            .build();
+
+        // send messages and consume them
+        clients.sendMessagesPlain();
+        clients.receiveMessagesPlain();
+
+        // save consumer group offsets
+        Map<String, String> offsetsBeforeBackup = clients.getCurrentOffsets();
+
+        // send additional messages
+        clients.setMessageCount(secondBatchSize);
+        clients.sendMessagesPlain();
+
+        // run backup procedure
+        LOGGER.info("Running backup procedure for {}/{}", NAMESPACE, clusterName);
+        String[] backupCommand = new String[] {
+            USER_PATH + "/../tools/cold-backup/run.sh", "-by", "-n", NAMESPACE, "-c", clusterName, "-t", backupFilePath
+        };
+        Exec.exec(true, backupCommand);
+
+        // delete the namespace and restore
+        ResourceManager.kubeClient().deleteNamespace(NAMESPACE);
+        NamespaceUtils.waitForNamespaceDeletion(NAMESPACE);
+        ResourceManager.kubeClient().createNamespace(NAMESPACE);
+        LOGGER.info("Running restore procedure for {}/{}", NAMESPACE, clusterName);
+        String[] restoreCommand = new String[] {
+            USER_PATH + "/../tools/cold-backup/run.sh", "-ry", "-n", NAMESPACE, "-c", clusterName, "-s", backupFilePath
+        };
+        Exec.exec(true, restoreCommand);
+
+        // deploy the operator and wait for provisionig
+        installClusterOperator(NAMESPACE);
+        LOGGER.info("Waiting for provisioning of restored {}/{}", NAMESPACE, clusterName);
+        KafkaUtils.waitForKafkaReady(clusterName);
+        clientsPodName = deployAndGetInternalClientsPodName();
+        clients = clients.toBuilder()
+            .withUsingPodName(clientsPodName)
+            .withMessageCount(secondBatchSize)
+            .build();
+
+        // check consumer group offsets
+        Map<String, String> offsetsAfterRestore = clients.getCurrentOffsets();
+        assertThat("Consumer group current offsets are same as before backup", offsetsAfterRestore, is(offsetsBeforeBackup));
+
+        // check consumer group recovery
+        assertThat("Consumer group is able to recover after restore", clients.receiveMessagesPlain(), is(secondBatchSize));
```

Comment:
When you add comment in the `assertThat`, it shows when the assert fails, so you should use inverse logic.
For example, if the assert fails, you will get `Consumer group is able to recover after restore`, which is not correct.

Reply:
Ok. Thanks.